Added steps to set proxy for vagrant

chandan5 opened the pull request:
If someone tries to setup development environment behind proxy he/she gets error when runs vagrant up because vagrant does not seem to obey system proxy or environment variables like HTTP_PROXY. Committing fix for adding proxy in vagrant.

smarx commented:
*Automated message from Dropbox CLA bot*

@chandan5, thanks for the pull request! It looks like you haven't yet signed the Dropbox CLA. Please [sign it here](https://opensource.dropbox.com/cla/) and update the thread so we can consider merging your code.

chandan5 commented:
I signed the Dropbox CLA. @smarx

timabbott commented:
Thanks for working on this @chandan5 !  I'm not enthusiastic about adding docs encouraging users to patch Vagrantfile, because Vagrantfile is in version control and it's kinda an annoying workflow to maintain a patched version of it while doing work on the project.  Is there another way we can provide the proxy configuration to Vagrant (e.g. via environment variables or an optional settings file that Vagrantfile can include?)

chandan5 commented:
Yeah I thought of getting environment variables like $http_proxy in Vagrantfile. Some Vagrantfile expert at zulip can maybe patch it. I tried from this stackoverflow [link](http://stackoverflow.com/questions/19648088/pass-environment-variables-to-vagrant-shell-provisioner)

On a separate note I was trying to share cloned zulip folder from Mac to Ubuntu in VirtualBox so that I can code faster on mac and use VM Ubuntu as a server. I found this [link](http://unix.stackexchange.com/questions/52667/file-permission-issues-with-shared-folders-under-virtual-box-ubuntu-guest-wind). I added my user to vboxsf group so that i can access files from shared zulip folder.
But when I run vagrant up on shared folder I get error 13 permission. Here is the full [log](http://pastebin.com/PSaciUeF)
I think some Vagrant user/process is trying to alter some mounted file. Currently only my user has permission to access mounted files. I also added lxc-dnsmasq hoping that might be the one. But I get the same error again.
Has anybody tried something like this ?
Any help would be great ?
P.S.: I have 2 VMs zulip is setup on 1 of them and I am trying this on other one.

timabbott commented:
Hmm, that's a bit confusing -- it sounds like you have a Mac, running an Ubuntu VM in virtualbox, and you're trying to run a Zulip LXC container inside that Ubuntu VM?  What problem are you trying to solve by running Zulip inside the container rather than just doing it directly in the Ubuntu Virtualbox VM?

I think this multilayer sharing is likely the cause of the permissions error you're getting...

chandan5 commented:
No I am not running multilayered vms.
I just tried to mount a shared folder from mac to ubuntu virtualbox vm and there were permission errors while running vagrant up.
Anyways If i figure it out, it would be nice. Meanwhile I have dev env setup on different vm with no shared folder.

timabbott commented:
Hmm, OK, well probably the issue has to do with the user that owns that shared folder being different from the user inside the VM.
